js/custom_lib.js

```diff
@@ -0,0 +1,260 @@
+ *
+ * @param {object} parent
+ * @param {String} tag
+ * @param {Array} attributes
+ * @return {object} The child element just created
+ */
+function createAndAppendChild(parent, tag, attributes) {
+  var child = createElementWithAttributes(tag, attributes);
+  parent.appendChild(child);
+  return child;
+}
+
+/**
+ * Create a HTML Element Tree from a js object.
+ *
+ * @example param
+ *  var lectureInfoTreeObj = {
+ *    tag: 'div',
+ *   class: 'lecture-info',
+ *    children: [
+ *      {
+ *        tag: 'input',
+ *        type: 'checkbox',
+ *        id: 'lecture-info-' + labelCount,
+ *        name: 'lecture-info'
+ *      },
+ *      {
+ *        tag: 'div',
+ *        class: 'lecture-info-header',
+ *        children: [
+ *          {
+ *            tag: 'label',
+ *            htmlFor: 'lecture-info-' + labelCount,
+ *            children: [
+ *              {
+ *                tag: 'div',
+ *                class: 'lecture-info-code',
+ *                innerHTML: (lecture.code + ' -')
+ *              }
+ *            ]
+ *          }
+ *        ]
+ *      }
+ *    ]
+ *  }
+ *
+ * @param rootObj
+ * @param {String} rootObj.tag HTML tag
+ * @param {String[]} rootObj.class CSS classes
+ * @param [rootObj.children] Objects like rootObj
+ * @return {HTML_Element} The root element
+ */
+function createHtmlElementTree(rootObj) {
+  if (!rootObj.tag) {
+    console.log('Error in createHtmlElementTree(', rootObj, ')');
+    return;
+  }
+
+  var tag = rootObj.tag;
+  delete rootObj.tag;
+  var children = rootObj.children;
+  delete rootObj.children;
+
+  var rootElement = createElementWithAttributes(tag, rootObj);
+
+  if (!children) {
+    return rootElement;
+  }
+
+  for (var i = 0; i < children.length; i++) {
+    var child = createHtmlElementTree(children[i]);
+    rootElement.appendChild(child);
+  }
+
+  return rootElement;
+}
+
+/**
+ * Translates 3-characters words in numbers.
+ * <br>
+ * 'seg' = 0, 'ter' = 1, ..., 'dom' = 6
+ *
+ * @param {String} day
+ * @return {Number}
+ */
+function indexOfDay(day) {
+  switch (day) {
+    case 'seg': return 0;
+    case 'ter': return 1;
+    case 'qua': return 2;
+    case 'qui': return 3;
+    case 'sex': return 4;
+    case 'sab': return 5;
+    case 'dom': return 6;
+  }
+}
+
+/**
+ * Creates a new array with unique elements from the array given.
+ *
+ * @param {Array} array
+ * @return {Array}
+ */
+function removeDuplicates(array) {
+  var seenHash = new Object();
+  var uniqueArray = new Array();
+  var uniqueCounter = 0;
+  for(var i = 0; i < array.length; i++) {
+    var item = array[i];
+    if(seenHash[item] != 1) {
+      seenHash[item] = 1;
+      uniqueArray[uniqueCounter++] = item;
+    }
+  }
+  return uniqueArray;
+}
+
+
+// TODO deixar essas funcoes (timeInMinutes, schedulesConflict, classroomsConflict) em plan.js ?
+//      a principio elas sao utilizadas **somente** la
```

Comment:
Be careful to avoid scope creep in plan.js. Personally, I'd prefer to see different libs for different areas like: date/time handling, schedule handling etc.